SimplenoteUITests/Generic.swift

```diff
@@ -0,0 +1,80 @@
+import XCTest
+
+var stepIndex = 0
+
+func attemptLogOut() -> Bool {
+    let allNotesNavBar = app.navigationBars[uidNavBar_AllNotes]
+    var loggedOut: Bool = false
+
+    if allNotesNavBar.exists {
+        loggedOut = logOut()
+    } else {
+        loggedOut = true
+    }
+
+    return loggedOut
+}
+
+func logOut() -> Bool {
+    app.navigationBars[uidNavBar_AllNotes].buttons[uidButton_Menu].tap()
+    app.tables.staticTexts[uidCell_Settings].tap()
+    app.tables.staticTexts[uidButton_Settings_LogOut].tap()
+    return app.buttons[uidButton_LogIn].waitForExistence(timeout: maxLoadTimeout)
+}
+
+public func trackTest(_ function: String = #function) {
+    print("> Test: \(function)")
+    stepIndex = 1
+}
+
+func logStep(){
+    print(">> Step " + String(stepIndex))
+    stepIndex += 1
+}
```

Comment:
Semi-related to these two functions to log the start of a test and its steps in the console: I recently learned about [`XCTActivity` and `XCTGroup`](https://developer.apple.com/documentation/xctest/activities_and_attachments/grouping_tests_into_substeps_with_activities) which sounds like it might be useful to us:

> Use activities to break longer test methods, such as UI tests and integration tests, into smaller named substeps. For example, break a complex UI login test into three substeps named "Open login window", "Enter password", and "Close login window", each of which contains the checks and assertions appropriate to that substep.